src/emqx_shared_sub.erl

```diff
@@ -75,16 +85,13 @@ start_link() ->
 subscribe(undefined, _Topic, _SubPid) ->
     ok;
 subscribe(Group, Topic, SubPid) when is_pid(SubPid) ->
-    mnesia:dirty_write(?TAB, record(Group, Topic, SubPid)),
-    gen_server:cast(?SERVER, {monitor, SubPid}).
+    mnesia:dirty_write(?TAB, ?SHARED_SUB(Group, Topic, SubPid)),
+    gen_server:cast(?SERVER, {subscribe, SubPid}).
 
 unsubscribe(undefined, _Topic, _SubPid) ->
     ok;
 unsubscribe(Group, Topic, SubPid) when is_pid(SubPid) ->
-    mnesia:dirty_delete_object(?TAB, record(Group, Topic, SubPid)).
-
-record(Group, Topic, SubPid) ->
-    #emqx_shared_subscription{group = Group, topic = Topic, subpid = SubPid}.
+    mnesia:dirty_delete(?TAB, {Group, Topic}).
```

Comment:
are you sure to discard the sub pid?
if a group is subscribed by multiple subscribers, one subscriber calling unsubscribe should not delete others.